### PR TITLE
feat(rules): add `no-jasmine-globals`

### PR DESCRIPTION
**Pull request by SimenB**
In preparation for jest-circus, warning users about Jasmine usage would be great.

Will probably add to recommended in next major (whenever eslint 5 comes out)

/cc @captbaritone

### CODE REVIEW

**Comment by aaronabramov** [comment]
oh. this is so awesome!

**Comment by mattphillips** [comment]
Looks like coverage has dropped slightly and is breaking CI. Maybe we should add coveralls to this project?

**Comment by SimenB** [comment]
Made `jasmine.DEFAULT_TIMEOUT` and known asymmetric matchers autofixable. not sure we can do any other ones

**Comment by captbaritone** [comment]
LGTM. 👏

**Comment by mattphillips** [comment]
@SimenB can we add a link to the docs for the recommended approach for the patterns we cannot auto fix?

**Comment by SimenB** [comment]
@mattphillips good idea! Feel free to push some examples to the docs 🙂

File: docs/rules/no-jasmine-globals.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+});
+```
+
+The following patterns would not be considered warnings:
+
+```js
+jest.setTimeout(5000);
+
+test('my test', () => {
+  spyOn(some, 'object');
```
Comment (thymikee) [inline docs/rules/no-jasmine-globals.md]: `spyOn` -> `jest.spyOn`

Comment (SimenB) [reply docs/rules/no-jasmine-globals.md]: hah :D

File: rules/__tests__/no-jasmine-globals.test.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+          line: 1,
+        },
+      ],
+    },
+    {
+      code: 'fail()',
+      errors: [
+        {
+          message:
+            'Avoid global `fail`, prefer throwing an error, or the `done` callback',
```
[the remark this answers is not in the document]
Comment (SimenB) [reply rules/__tests__/no-jasmine-globals.test.js]: good call

File: rules/__tests__/no-jasmine-globals.test.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+          column: 1,
+          line: 1,
+        },
+      ],
+    },
+    {
+      code: 'pending()',
+      errors: [
+        {
+          message: 'Avoid global `pending`, prefer explicitly skipping a test',
```
[the remark this answers is not in the document]
Comment (SimenB) [reply rules/__tests__/no-jasmine-globals.test.js]: yeeeeah

**Comment by SimenB** [comment]
Changed "avoid using" to "illegal usage of"

**Comment by thymikee** [comment]
Like it 👍

**Comment by thymikee** [comment]
:shipit: when green :)

**Comment by captbaritone** [comment]
@aaronabramov We should turn this on for WWW once you are done.

**Comment by captbaritone** [comment]
Should this get added to the readme? If so, I can open a PR.

**Comment by thymikee** [comment]
Of course! Maybe we can even make it default?

**Comment by captbaritone** [comment]
I'll open a PR.

**Comment by captbaritone** [comment]
I'll wait on making it recommended since @SimenB  said:

> Will probably add to recommended in next major (whenever eslint 5 comes out)

**Comment by SimenB** [comment]
Found a bug btw, when applying it to the jest code base. Pushed fix: 885ce1798d9b0bd5b6685675f8456062d34fc551

**Comment by SimenB** [comment]
Interesting that the bot picked up I referenced the commit in this PR, so posted a comment. Probably a bug :P

**Comment by SimenB** [comment]
For posterity, these are the errors reported on the Jest code base with this plugin active:
```
/Users/simen/Development/jest/packages/jest-jasmine2/src/index.js
44:15  error  Illegal usage of jasmine global  jest/no-jasmine-globals

/Users/simen/Development/jest/packages/jest-jasmine2/src/jasmine_async.js
136:15  error  Illegal usage of jasmine global  jest/no-jasmine-globals

/Users/simen/Development/jest/packages/jest-jasmine2/src/jest_expect.js
36:3  error  Illegal usage of jasmine global  jest/no-jasmine-globals
37:3  error  Illegal usage of jasmine global  jest/no-jasmine-globals
38:3  error  Illegal usage of jasmine global  jest/no-jasmine-globals
39:3  error  Illegal usage of jasmine global  jest/no-jasmine-globals
40:3  error  Illegal usage of jasmine global  jest/no-jasmine-globals
42:3  error  Illegal usage of jasmine global  jest/no-jasmine-globals

/Users/simen/Development/jest/testSetupFile.js
16:3  error  Illegal usage of jasmine global  jest/no-jasmine-globals
```

**Pull request merged**